### PR TITLE
Use config override at README examples

### PR DESCRIPTION
**Pull request by hasancelik**
`TLS_ENABLED` -- > `HZ_NETWORK_SSL_ENABLED `
`HZ_LICENSE_KEY` --> `HZ_LICENSEKEY`
`JAVA_OPTS="-Dhazelcast.local.publicAddress` --> `HZ_NETWORK_PUBLICADDRESS`

### CODE REVIEW

**Review by pivovarit — APPROVED** [review]

**Review by leszko — COMMENTED** [review]
Is this change backward-compatible?

**Comment by hasancelik** [comment]
> Is this change backward-compatible?

@leszko config override introduced with 4.1 release so `HZ_NETWORK_SSL_ENABLED, HZ_LICENSEKEY, HZ_NETWORK_PUBLICADDRESS` are not avaliable for previous releases like 4.0.x . As a result, it's not backward-compatible.

**Comment by leszko** [comment]
> > Is this change backward-compatible?
> 
> @leszko config override introduced with 4.1 release so `HZ_NETWORK_SSL_ENABLED, HZ_LICENSEKEY, HZ_NETWORK_PUBLICADDRESS` are not avaliable for previous releases like 4.0.x . As a result, it's not backward-compatible.

Then I guess we need to tweak it, so that the old env variables would still work correctly.

**Comment by pivovarit** [comment]
We could just provide two versions? one for >4.1 and one for 4.0.x?

**Comment by leszko** [comment]
> We could just provide two versions? one for >4.1 and one for 4.0.x?

What do you mean by 2 versions? I think we need to check the old variable and if it's defined then assign it to the new variable. For the sake of backward-compatibility.

**Comment by pivovarit** [comment]
Hmm, now I see you asked about something else. This should definitely stay backward compatible.

I just meant explanations in docs that could be misleading for older images (since there's no support for the env-based config)

**Comment by hasancelik** [comment]
@leszko @pivovarit What about creating a new branch for 4.0.z images?

**Comment by leszko** [comment]
@hasancelik why would we need such a branch?

About the docs, I think the docs at master should be about the current master version. If someone needs docs for older images, then they can check the tagged README.

My point about the backward-compatibility was just that these commands should both work now.
```
$ docker run -e HZ_LICENSE_KEY=<license-key> hazelcast/hazelcast-enterprise
$ docker run -e HZ_LICENSEKEY=<license-key> hazelcast/hazelcast-enterprise
```

**Comment by hasancelik** [comment]
> About the docs, I think the docs at master should be about the current master version. If someone needs docs for older images, then they can check the tagged README.

@leszko with next `4.0.x` release, HZ_VERSION will be updated to `4.0.4` and related image/tag will be created from master branch by IMDG release script. This tag includes same README file with `4.1.1` so release management process of docker images will become a little bit complicated. If we decide to continue with only master, we need to provide two command/explanation for 4.0.x and 4.1.x at README.

**Comment by leszko** [comment]
> > why would we need such a branch?
> 
> > About the docs, I think the docs at master should be about the current master version. If someone needs docs for older images, then they can check the tagged README.
> 
> @leszko with next `4.0.x` release, HZ_VERSION will be updated to `4.0.4` and related image/tag will be created from master branch by IMDG release script. This tag includes same README file with `4.1.1` so release management process of docker images will become a little bit complicated. If we decide to continue with only master, we need to provide two command/explanation for 4.0.x and 4.1.x at README.

Ok, then, yes, we should create the branch you suggested.

File: README.md [hunk 1]
```diff
@@ -132,6 +107,34 @@ $ docker run -m 512M hazelcast/hazelcast:$HAZELCAST_VERSION
...
+hazelcast:
+  cluster-name: dev
+  network:
+    port:
+      auto-increment: true
+      port-count: 100
+      port: 5701
+    join:
+      auto-detection:
+        enabled: true
```
Comment (leszko) [inline README.md:125]:
```suggestion
hazelcast:
  cluster-name: dev
  network:
    port:
      auto-increment: true
      port-count: 100
      port: 5701
```

Auto-detection is enabled by default, so it does not make much sense to use it in the example.

File: README.md [hunk 2]
```diff
@@ -132,6 +107,34 @@ $ docker run -m 512M hazelcast/hazelcast:$HAZELCAST_VERSION
...
+```
+
+If you want to use the environment variables, the above would be represented as a set of the following environment variables:
+```bash
+$ docker run -e HZ_CLUSTERNAME=dev \
+  -e HZ_NETWORK_PORT_AUTOINCREMENT=true \
+  -e HZ_NETWORK_PORT_PORTCOUNT=100 \
+  -e HZ_NETWORK_PORT_PORT=5701 \
+  -e HZ_NETWORK_JOIN_AUTODETECTION_ENABLED=true \
+  hazelcast/hazelcast
```
Comment (leszko) [inline README.md:135]:
```suggestion
$ docker run -e HZ_CLUSTERNAME=dev \
  -e HZ_NETWORK_PORT_AUTOINCREMENT=true \
  -e HZ_NETWORK_PORT_PORTCOUNT=100 \
  -e HZ_NETWORK_PORT_PORT=5701 \
  hazelcast/hazelcast
```

File: README.md [hunk 3]
```diff
@@ -152,6 +155,40 @@ $ docker run -e CLASSPATH="/opt/hazelcast/CLASSPATH_EXT/*" -v /home/ubuntu/hazel
 
 Alternatively, you can [extend Hazelcast base image](#extending-hazelcast-base-image) adding your custom JARs.
 
+
+### Using TLS (Hazelcast Enterprise Only)
+
+The `HZ_NETWORK_SSL_ENABLED` environment variable can be used to enable TLS for the communication. You can check TSL related section at the documantation, [here](https://docs.hazelcast.org/docs/4.1.1/manual/html-single/index.html#integrating-openssl-boringssl).
```
Comment (leszko) [inline README.md:161]: I guess what you describe here (and in point 1) is related to OpenSSL, which we actually don't recommend. I guess point 2 should be enough to use TLS with the standard Java TLS implementation.

Comment (hasancelik) [reply README.md]: yes, you are totally right, done 👍

**Review by leszko — APPROVED** [review]
LGTM 👍 

Sorry for such a late review.

@pivovarit could you please look and approve as well?

**Review by pivovarit — APPROVED** [review]
LGTM

**Pull request merged**